src/core/Badge/Badge.tsx

```diff
@@ -7,13 +7,48 @@ import cx from 'classnames';
 import { CommonProps } from '../utils/props';
 import { useTheme } from '../utils/hooks/useTheme';
 import '@itwin/itwinui-css/css/badge.css';
+import { getWindow, isSoftBackground, SoftBackgrounds } from '../utils/common';
+
+/**
+ * Helper function that returns one of the preset badge color values.
+ */
+const getBadgeColorValue = (color: BadgeProps['backgroundColor']) => {
+  if (!color) {
+    return '';
+  }
+
+  switch (color) {
+    case 'primary':
+      return '#A5D7F5';
+    case 'positive':
+      return '#C3E1AF';
+    case 'negative':
+      return '#EFA9A9';
+    case 'warning':
+      return '#F9D7AB';
+    default:
+      if (isSoftBackground(color)) {
```

Comment:
Wouldn't it be simpler just by doing:
```ts
return SoftBackgrounds[color] ?? color;
```

Reply:
The type guard is needed to prevent typescript error